libnopegl/src/node_drawother.c

```diff
@@ -580,63 +579,59 @@ static int init(struct ngl_node *node,
     s->uvcoord_attr.format = NGLI_FORMAT_R32G32_SFLOAT;
 
     if (!o->geometry) {
-        s->uvcoords = ngli_buffer_create(gpu_ctx);
-        s->vertices = ngli_buffer_create(gpu_ctx);
-        if (!s->uvcoords || !s->vertices)
+        s->own_geometry = 1;
+
+        s->geometry = ngli_geometry_create(gpu_ctx);
+        if (!s->geometry)
             return NGL_ERROR_MEMORY;
 
-        int ret;
-        if ((ret = ngli_buffer_init(s->vertices, sizeof(default_vertices), VERTEX_USAGE_FLAGS)) < 0 ||
-            (ret = ngli_buffer_init(s->uvcoords, sizeof(default_uvcoords), VERTEX_USAGE_FLAGS)) < 0 ||
-            (ret = ngli_buffer_upload(s->vertices, default_vertices, 0, sizeof(default_vertices))) < 0 ||
-            (ret = ngli_buffer_upload(s->uvcoords, default_uvcoords, 0, sizeof(default_uvcoords))) < 0)
+        int ret = ngli_geometry_set_vertices(s->geometry, 4, default_vertices);
+        if (ret < 0)
             return ret;
 
-        s->position_attr.stride = 3 * sizeof(float);
-        s->position_attr.buffer = s->vertices;
+        ret = ngli_geometry_set_uvcoords(s->geometry, 4, default_uvcoords);
+        if (ret < 0)
+            return ret;
 
-        s->uvcoord_attr.stride = 2 * sizeof(float);
-        s->uvcoord_attr.buffer = s->uvcoords;
+        ret = ngli_geometry_init(s->geometry, NGLI_PRIMITIVE_TOPOLOGY_TRIANGLE_STRIP);
+        if (ret < 0)
+            return ret;
```

Comment:
Style: can we keep the existing compact pattern?

```c
        int ret;
        if ((ret = ngli_geometry_set_vertices(s->geometry, 4, default_vertices)) < 0 ||
            (ret = ngli_geometry_set_uvcoords(s->geometry, 4, default_uvcoords)) < 0 ||
            (ret = ngli_geometry_init(s->geometry, NGLI_PRIMITIVE_TOPOLOGY_TRIANGLE_STRIP) < 0))
            return ret;
```

Reply:
I'm not huge found of that style because the day you need to add some error logging you need to unwrap the whole pattern. For now, I can keep the style this way.